score.go

```diff
@@ -334,6 +376,45 @@ func (ps *peerScore) inspectScores() {
 	go ps.inspect(scores)
 }
 
+func (ps *peerScore) inspectScoresExtended() {
+	ps.Lock()
+	scores := make(map[peer.ID]*PeerScoreSnapshot, len(ps.peerStats))
+	for p, pstats := range ps.peerStats {
+		pss := new(PeerScoreSnapshot)
+		pss.Score = ps.score(p)
+		if len(pstats.topics) > 0 {
+			pss.Topics = make(map[string]*TopicScoreSnapshot, len(pstats.topics))
+			for t, ts := range pstats.topics {
+				pss.Topics[t] = &TopicScoreSnapshot{
+					FirstMessageDeliveries:   ts.firstMessageDeliveries,
+					MeshMessageDeliveries:    ts.meshMessageDeliveries,
+					InvalidMessageDeliveries: ts.invalidMessageDeliveries,
+				}
+				if ts.inMesh {
+					pss.Topics[t].TimeInMesh = ts.meshTime
+				}
+			}
+		}
+		pss.AppSpecificScore = ps.params.AppSpecificScore(p)
+		for _, ip := range pstats.ips {
+			_, whitelisted := ps.params.IPColocationFactorWhitelist[ip]
+			if whitelisted {
+				continue
+			}
+			peersInIP := len(ps.peerIPs[ip])
+			if peersInIP > ps.params.IPColocationFactorThreshold {
+				surpluss := peersInIP - ps.params.IPColocationFactorThreshold
+				pss.IPColocationFactor += surpluss
+			}
```

Comment:
Isn't this the code to _actually_ calculate the score? If so, we should encapsulate it in a function, so it's not duplicated and there's no risk of falling out of sync if we ever update the logic.

Reply:
yeah, I can extract to a function.

Reply:
done.